### ISSUE TITLE
No coffee

### ISSUE THREAD

**Issue by bilalq** [exchgr/SlamWhale#17]
Would you guys be opposed to Coffeescript?

**Comment by x** [comment]
I would be opposed. I don't believe in Coffeescript.

**Comment by bilalq** [comment]
:(

@exchgr, @MrJaeger, any thoughts?

**Comment by exchgr** [comment]
I believe in the wonders of Coffeescript.

**Comment by bilalq** [comment]
I went ahead and rewrote everything in Coffee. However, the code was already pretty clean, and since we're not doing much in the way of functional programming, I didn't see much of a benefit. I am now of the opinion that it isn't worth complicating our build process to bring in Coffee for this project.

However, I do want to revisit our style here. Can we all agree to stop using semi-colons?

**Comment by exchgr** [comment]
Just wondering, how does Coffee affect the build process? Isn't it possible to just run it directly without compiling it to JavaScript?

Also, if we use Coffee, we never have to write semicolons, nor do we have to write curly braces or unnecessary parentheses. We also get some goodness that JavaScript doesn't necessarily have, like `for...in` loops.

**Comment by bilalq** [comment]
For the server side, code, yes, it is indeed possible to run directly. However, we'd need to bring in Grunt or Gulp to handle the client-side code. This may not be that bad though. If we start using compass, I'd consider it worthwhile.

**Comment by x** [comment]
I really think we would be unnecessarily complicating this project. Especially considering how simple the front-end is.

We've already proven that we have issues maintaining our motivation for this project. I think anything that motivates us to write code is worth doing, so if you're super motivated to rewrite it in Coffee, do it.

But if we just have raw motivation for the project (like I had when I rewrote it), we should choose the shortest path towards improvement, which I believe is bug fixes and new features.

**Comment by exchgr** [comment]
I agree with Devon. Let's focus on the important stuff: moving SlamWhale forward.
